var/spack/repos/builtin/packages/curl/package.py

```diff
@@ -48,204 +47,18 @@ class Curl(NMakePackage, AutotoolsPackage):
         sha256="9b6b1e96b748d04b968786b6bdf407aa5c75ab53a3d37c1c8c81cdb736555ccf",
         deprecated=True,
     )
+    # needed by old r
```

Comment:
```suggestion
    # needed by old r
```

I might suggest adding the specific versions of `r` requiring `curl@7.88.1` to the comment so we'll know we can prune this too when that version of `r` is deprecated and removed later.

Reply:
i don't see how that helps, it's just a comment.

Reply:
Just one fewer scavenger hunt later on. This way in a year or so when we want to clean up versions again we can see if `r@:4.2` versions still exist or if they've all been removed. Else we'd have to open up `r/package.py` and then read through the list of `versions` and `depends_on` directives to see if the versions were removed. 

And if someone removed both the versions and the depends directives we'd have to dig into the history to verify that we can remove the version of curl.

We could also automate this with an audit like script that'd be pretty neat!